frontend/pages/hosts/details/cards/Software/HostSoftwareTable/HostSoftwareTable.tsx

```diff
@@ -96,9 +96,29 @@ const HostSoftwareTable = ({
     [determineQueryParamChange, pagePath, generateNewQueryParams, router]
   );
 
+  const getItemsCountText = () => {
+    const count = data?.count;
+    if (!data?.software?.length || !count) return "";
+
+    return count === 1 ? `${count} software item` : `${count} software items`;
+  };
+
+  const renderSoftwareCount = () => {
+    const itemText = getItemsCountText();
+
+    if (!itemText) return null;
+
+    return (
+      <div className={`${baseClass}__count`}>
+        <span>{itemText}</span>
+      </div>
+    );
+  };
+
   return (
     <div className={baseClass}>
       <TableContainer
+        renderCount={renderSoftwareCount}
```

Comment:
We might want useMemo/useCallback to keep the function references stable and avoid unnecessary re-renders of TableContainer (although this is an issue more generally, for example I think in this component emptyComponent is also not a stable across renders from TableContainer's perspective). We can keep an eye on it and change up the approach if we see performance issues.